aoc2022/src/main/scala/day08.scala

```diff
@@ -0,0 +1,76 @@
+import scala.io.*
+import math.*
+import scala.annotation.tailrec
+
+/**
+ * PART 01:
+ *
+ * The solution for me was to have two Vector[Vector[Int]] available: one with the input as is, and one with the input 
+ * transposed. This allowed me to just have to check the left and right side of a tree, in the original input and in 
+ * the transposed input. The transposed input essentially are the trees above and below the current tree. 
+ *
+ * PART 02:
+ *
+ * Refactored the isVisible() function from part 1 to now compute the score. It still works using the original input and 
+ * the transposed input. 
+ *
+ */
+
+
+object day08 extends App:
+
+  private val day: String =
+    this.getClass.getName.drop(3).init
+
+  private val start1: Long =
+    System.currentTimeMillis
+
+  private val input: Vector[TreeRow] =
+    Source
+      .fromResource(s"day$day.txt")
+      .getLines
+      .toVector
+      .map(_.map(_.toString.toInt).toVector)
+
+
+  type TreeRow = Vector[Int]
+
+  def isVisible(i: Int, j: Int, trees: TreeRow, treesTransposed: TreeRow): Boolean =
+    def checkRow(index: Int, row: TreeRow): Boolean =
+      val (left, right): (TreeRow, TreeRow) = row.splitAt(index)
+      if left.forall(_ < row(index)) || right.drop(1).forall(_ < row(index)) then true
+      else false
+
+    if i == 0 || i == trees.length then true                  // tree is on border and thus visible
+    else if j == 0 || j == treesTransposed.length then true   // tree is on border and thus visible
```

Comment:
Could there be an off-by-one error here? If I understand correctly `i` and `j` are indices and are always strictly less than the length...